java-source/src/main/java/com/example/state/IOUState.java

```diff
@@ -62,7 +61,7 @@ public IOUState(IOU iou,
      */
     @Override public boolean isRelevant(Set<? extends PublicKey> ourKeys) {
         final List<PublicKey> partyKeys = Stream.of(sender, recipient)
-                .flatMap(party -> party.getOwningKey().getKeys().stream())
+                .map(Party::getOwningKey)
```

Comment:
`isRelevant` function is tricky. After the change in `CompositeKey/PublicKey` we have to remember, that `PublicKey` can still be a `CompositeKey`, so if we want to collect all the leaf keys, we still need to run `getKeys()` (which type checks if it is `CompositeKey`).